### PR TITLE
Enable IPv6 support for BK72xx

### LINKED ISSUE #2853: BK72xx IPv6 support
**Describe the problem you have/What new integration you would like**
IPv6 support on BK72xx platform

**Please describe your use case for this integration and alternatives you've tried:**
BK72xx only has Legacy IP support.

**Additional context**
It was disabled in https://github.com/esphome/esphome/pull/6387 without any comment about why it didn't work, or links to upstream libretiny bugs to track.

### PR DESCRIPTION
**Pull request by dwmw2**
Requires:
- [x] LibreTiny-LwIP: https://github.com/libretiny-eu/lwip/pull/2
- [x] LibreTiny: https://github.com/libretiny-eu/libretiny/pull/292
  - [x] And a release numbered at least 1.7.0
- [x] AsyncTCP-esphome: https://github.com/esphome/AsyncTCP/pull/14
  - [x] AsyncTCP-esphome: https://github.com/esphome/AsyncTCP/pull/15
  - [x] ESPHome: https://github.com/esphome/esphome/pull/7431
- [x] ESPHome: https://github.com/esphome/esphome/pull/7408
- [x] ESPHome: https://github.com/esphome/esphome/pull/7409
- [x] ESPHome: https://github.com/esphome/esphome/pull/7410
- [x] ESPHome: https://github.com/esphome/esphome/pull/7415

Closes https://github.com/esphome/feature-requests/issues/2853

# What does this implement/fix?

Add IPv6 support for BK72xx

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes https://github.com/esphome/feature-requests/issues/2853

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [x] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
network:
  enable_ipv6: true
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).
 (No, because it looked like https://esphome.io/components/network.html never did specify that IPv6 was only available on some platforms anyway)

### CODE REVIEW

**Comment by dwmw2** [comment]
Will also need to pull in the newer AsyncTCP with https://github.com/esphome/AsyncTCP/pull/14 fixed.

**Comment by dwmw2** [comment]
No longer waiting for upstream; just for the ESPHome PRs #7409 and #7410.
_(Those could possibly have been included in *this* PR, but I've taken the advice "always separate PRs" literally.)_

**Comment by dwmw2** [comment]
@kuba2k2 It looks like the bots dropped 'integration:libretiny' from this, although it still very much is for LibreTiny.

**Review by kuba2k2 — APPROVED** [review]

File: esphome/components/async_tcp/__init__.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 async def to_code(config):
     if CORE.is_esp32 or CORE.is_libretiny:
         # https://github.com/esphome/AsyncTCP/blob/master/library.json
-        cg.add_library("esphome/AsyncTCP-esphome", "2.1.3")
+        cg.add_library("esphome/AsyncTCP-esphome", "2.1.4")
```
Comment (jesserockz) [inline esphome/components/async_tcp/__init__.py]: Please split this into its own PR, and also needs to be bumped in `platformio.ini`

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/async_tcp/__init__.py (hunk 1 above)
Comment (dwmw2) [reply esphome/components/async_tcp/__init__.py]: Done,. thanks.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**